src/dos/drive_fat.cpp

```diff
@@ -1023,22 +1045,26 @@ bool fatDrive::FileCreate(DOS_File **file, char *name, uint16_t attributes) {
 		if(!getDirClustNum(name, &dirClust, true)) return false;
 		memset(&fileEntry, 0, sizeof(direntry));
```

Comment:
This zero'ing a bit fragile (we might as well improve it a bit while we're here):

`memset(&fileEntry, 0, sizeof(direntry));`

It's fragile because the variable and the type might be different (there's not guarantee fileEntry isn't so other type).

But we can make this even easier to read with:

`fileEntry = {};`

To see that in action:

https://onlinegdb.com/Jqk_zZL9E